### PR TITLE
DOC: @everywhere and adding processors

### PR DESCRIPTION
**Pull request by stsievert**
I'm a new Julia user and had been trying to adding processes. It took me a while to realize that `@everywhere` only defines processes on the processors that have already been added to julia.

This PR helps with this documentation issue. It adds a note in `@everywhere`.

### CODE REVIEW

File: base/multi.jl [hunk 1]
```diff
@@ -1981,6 +1981,9 @@ For example :
     FooBar.foo()
 
 will result in `Main.bar` being defined on all processes and not `FooBar.bar`.
+
+The expression is evaluated on all current processes. Any processes added later
+(say with `addprocs`) will not have the expression defined.
```
Comment (kshyatt) [inline base/multi.jl]: Could have a link here to `addprocs`. You want 

```julia
[`addprocs`](:func:`addprocs`)
```

**Comment by kshyatt** [comment]
LGTM. Thanks for the doc improvement, and welcome to Julia!

**Comment by stsievert** [comment]
Rebased and added the link.

**Comment by stsievert** [comment]
Can we merge this?

File: base/multi.jl [hunk 2]
```diff
@@ -1981,6 +1981,9 @@ For example :
     FooBar.foo()
 
 will result in `Main.bar` being defined on all processes and not `FooBar.bar`.
+
+The expression is evaluated on all current processes. Any processes added later
+(say with [`addprocs`](:func:`addprocs`)) will not have the expression defined.
```
Comment (tkelman) [inline base/multi.jl]: the cross reference format has changed, I think it's just `(@ref)` now

Comment (stsievert) [inline base/multi.jl]: Added, I used [stdlin/collections.md][1]  as an example. 

    [`addprocs`](:func:`addprocs`) => [`addprocs()`](@ref)

[1]:https://github.com/JuliaLang/julia/blob/7b7b4e1d24aa449bf33af93e684ca8924b66a010/doc/src/stdlib/collections.md

**Comment by ViralBShah** [comment]
This looks good. Seems like we missed it back then, Would be nice to get it in, either by fixing the conflict, or doing a new PR.

**Comment by stsievert** [comment]
I've resolved the merge (the declaration of `everywhere` moved).

File: base/multi.jl [hunk 3]
```diff
@@ -0,0 +1,2257 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
```
Comment (fredrikekre) [inline base/multi.jl:1]: I don't think you meant to commit this file.

Comment (stsievert) [reply base/multi.jl]: Whoops! Thanks. Now the diff is much smaller than 2000 lines.

**Pull request merged**